.github/workflows/ci.yaml

```diff
@@ -12,23 +12,23 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@master
-      - uses: actions/cache@v2.1.6
+      - uses: actions/checkout@v4.2.2
+      - uses: actions/cache@v4.1.2
         with:
           path: ~/go/pkg/mod
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
-      - uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.0
+      - uses: actions/setup-go@v5.1.0
```

Comment:
Can we use the sha for all the actions instead of the tag?

Reply:
@rumstead yes.  sha definitely more secure.  will update.